internal/controller/driver_controller.go

```diff
@@ -105,5 +164,156 @@ func (r *driverReconcile) LoadAndValidateDesiredState() error {
 		return client.IgnoreNotFound(err)
 	}
 
+	// Creating a copy of the driver spec, making sure any local changes will not effect the object residing
+	// in the client's cache
+	r.driver.Spec = *r.driver.Spec.DeepCopy()
+	mergeDriverSpecs(&r.driver.Spec, opConfig.Spec.DriverSpecDefaults)
+	mergeDriverSpecs(&r.driver.Spec, &driverDefaults)
+
 	return nil
 }
+
+// mergeDriverSpecs will fill in any unset fields in dest with a copy of the same field in src
+func mergeDriverSpecs(dest, src *csiv1a1.DriverSpec) {
+	// Create a copy of the src, making sure that any value copied into dest is a not shared
+	// with the original src
+	src = src.DeepCopy()
```

Comment:
we can make use of the reflect package to get all the values of struct
`valResult := reflect.ValueOf(&result).Elem()`

And simply run the for loop on `valResult.NumField()`

So we don't need to do this individual, and not maintain this merge function if any field is added in future

Reply:
@parth-gr 
Quoting the creator of GO, reflection is never simple or readable. And It is also not performed.

Trying to solve a simple problem with an over-generalized solution isn't necessary and might even be harmful. As an explanation of the previous statement please consider the following:
- The code is very simple as is, any junior engineer can read and understand it. Moving to a reflection-based approach will most definitely result in future misunderstanding and potential bugz
- The problem here is not a meta-programming problem, so choosing reflection is the wrong tool for the job. 
- It is not evident that every current and future field will fall into a common generalization (or in other words, will have a similar merging logic). Assuming otherwise is an analysis fallacy. If we write a reflection-based code we will most definitely have exceptions. That, in turn, will result in an even more complicated code to handle them. 

As a rule of thumb, the simpler the code the more supportable and maintainable it is, and has less surface area for non-obvious bugs.  I will always prioritize simplicity and readability over "smart" solutions (or more likely overcomplicated ones.

Reply:
@nb-ohad 
Make sense that this bring a addon complexity but the benefit it brought is not negligible, 

> It removes the dependency to not touch this `merge code`, whenever there is an API change
> obviously reduces the one file code to one small function
> for the exceptions we can have a separate function which distinguish which spec follows different pattern.

Any way its your call, nothing urgent but a good point of improvement.

Reply:
> It removes the dependency to not touch this `merge code`, whenever there is an API change
@parth-gr I will argue that we want an engineer to touch that code whenever the API changes. Because it forces the engineer to think about the proper merge logic.
Case in point your log rotation changes, and how they changed because of merging considerations.

> obviously reduces the one file code to one small function
This should never be a metric for anything. If the file size really offend you, we can always move this code into a different file. 

> for the exceptions we can have a separate function which distinguish which spec follows different pattern.
This is exactly the unwanted complexity I was talking about, as now whoever needs to work on this code and hold a mental mode of this code will need to consider that as well, and it is not an obvious consideration, especially when comparing it to just have a function that has a bunch of simple ifs.  Regarding any concerns that someone will add an API field but will not add the merge code, I will ask to wait until I will provide the proper test suite for this function (which will cover that concern exactly) 

To all of the above I will add that `reflection` is really hard to debug with any and all of the existing golang debuggers

Reply:
Okay thanks